lib/web_of_science/contributions.rb

```diff
@@ -76,30 +76,11 @@ def found_contribution?(author, record)
     # @param value [String]
     # @return [Boolean] contribution exists
     def contribution_by_identifier?(author, type, value)
-      contrib = contribution_by_identifier(author, type, value)
+      return false if type.blank? || value.blank?
+      pub_id = PublicationIdentifier.find_by(identifier_type: type, identifier_value: value)
+      return false if pub_id.nil?
+      contrib = find_or_create_contribution(author, pub_id.publication)
       contrib.nil? ? false : contrib.persisted?
     end
-
-    # Find any matching contribution by author and PublicationIdentifier
-    # @param author [Author]
-    # @param type [String]
-    # @param value [String]
-    # @return [Contribution, nil]
-    def contribution_by_identifier(author, type, value)
-      pub_id = publication_identifier(type, value)
-      return if pub_id.nil?
-      find_or_create_contribution(author, pub_id.publication)
-    end
-
-    # Is there a PublicationIdentifier matching the type and value?
-    # @param type [String]
-    # @param value [String]
-    # @return [PublicationIdentifier, nil]
-    def publication_identifier(type, value)
-      return if type.blank? || value.blank?
-      pub_ids = PublicationIdentifier.where(identifier_type: type, identifier_value: value)
-      pub_ids.empty? ? nil : pub_ids.first
-    end
-
   end
```

Comment:
- At some point, the original version of this method was something like this.  To fix the code-climate complexity, it was broken out into component methods.  In doing so, it focuses each method on a specific purpose.
- I need to double check whether any of these methods are used elsewhere.
- Not entirely sure this buys significant performance improvement (apart from a bit of method call overhead)

Reply:
- Maybe the `Publication.includes(:publication_identifiers).find_by` is a performance gain?
- Perhaps performance improvements would be better gained by looking at indexes on the PublicationIdentifier values (and cleaning up the data in that table, along with validations).

Reply:
(BTW, going through this commit-by-commit, not sure how github will display the resulting comments.  Seems like some commits on the Contributions might be squashed, if all this is consolidated and methods removed are not used elsewhere, and code-climate complexity is ignored.)

Reply:
Re: `Publication.includes(:publication_identifiers).find_by`, yes.  We ask for the thing we want rather than searching for the pub ID and then asking for `pub_id.publication` (which causes a separate DB query).